airflow/models/dagrun.py

```diff
@@ -498,6 +498,37 @@ def fetch_task_instances(
             tis = tis.where(TI.task_id.in_(task_ids))
         return session.scalars(tis).all()
 
+    @internal_api_call
+    def _check_last_N_dagruns_failed(
+        self,
+        dag_id: str,
+        number_of_dag_runs: int,
+        session: Session
+    ):
+        """Check if last N dags failed."""
+        dag_runs = session.query(DagRun).filter(DagRun.dag_id == dag_id) \
+            .order_by(DagRun.execution_date.desc()) \
+            .limit(number_of_dag_runs).all()
+
+        """ Marking dag as paused, if needed"""
+        toBePaused = all(dag_run.state == DagRunState.FAILED for dag_run in dag_runs)
```

Comment:
To ensure the correct behavior, we need to check the count of failures as well.

Reply:
We are checking last N dagruns and checking if all the failed,
this will make sure we have the required count.
please highlight if you think otherwise.

Reply:
I am referring to the limit query might give less records. So we need cross check the failed count.

Reply:
ohh, yeah
thanks for pointing it out.